src/Hedgehog/Property.fs

```diff
@@ -78,8 +81,11 @@ module Property =
             | Success x ->
                 GenTuple.mapFst (Journal.append journal) (k x))
 
+    let handle (e : exn) =
+        (Journal.singletonMessage (string e), Failure) |> Gen.constant
+
     let bind (k : 'a -> Property<'b>) (m : Property<'a>) : Property<'b> =
-        bindGen (toGen << k) (toGen m) |> ofGen
+        m |> toGen |> bindGen (fun a -> (try k a |> toGen with e -> handle e)) |> ofGen
```

Comment:
There's enough going on with this line to warrant multiple lines.

Reply:
Also, I'd extract the lambda to an inner function so the `try`/`with` can be on separate lines as well.

Reply:
Done